staging/src/k8s.io/code-generator/cmd/prerelease-lifecycle-gen/prerelease-lifecycle-generators/status.go

```diff
@@ -0,0 +1,447 @@
+		With("introducedMinor", introducedMinor)
+
+	// if someone intentionally override the deprecation release
+	if tagExists(deprecatedTagName, t) {
+		tag, tagMajor, tagMinor, err := extractDeprecatedTag(t)
+		if err != nil {
+			return nil, err
+		}
+		a = a.With("deprecated", tag.value).
+			With("deprecatedMajor", tagMajor).
+			With("deprecatedMinor", tagMinor)
+
+	} else { // if not overridden, compute based on our policy
+		tagMajor := introducedMajor
+		tagMinor := introducedMinor + 3
+		a = a.With("deprecated", fmt.Sprintf("%d.%d", tagMajor, tagMinor)).
+			With("deprecatedMajor", tagMajor).
+			With("deprecatedMinor", tagMinor)
+	}
+
+	// if someone intentionally override the removed release
+	if tagExists(removedTagName, t) {
+		tag, tagMajor, tagMinor, err := extractRemovedTag(t)
+		if err != nil {
+			return nil, err
+		}
+		a = a.With("removed", tag.value).
+			With("removedMajor", tagMajor).
+			With("removedMinor", tagMinor)
+
+	} else { // if not overridden, compute based on our policy
+		tagMajor := introducedMajor
+		tagMinor := introducedMinor + 6
+		a = a.With("removed", fmt.Sprintf("%d.%d", tagMajor, tagMinor)).
+			With("removedMajor", tagMajor).
+			With("removedMinor", tagMinor)
+	}
+
+	return a, nil
+}
+
+func (g *genPreleaseLifecycle) Init(c *generator.Context, w io.Writer) error {
+	return nil
+}
+
+func (g *genPreleaseLifecycle) GenerateType(c *generator.Context, t *types.Type, w io.Writer) error {
+	klog.Infof("Generating prerelease-lifecycle for type %v", t)
+
+	sw := generator.NewSnippetWriter(w, c, "$", "$")
+	args, err := argsFromType(t)
+	if err != nil {
+		return err
+	}
+
+	if versionedMethodOrDie("Introduced", t) == nil {
+		sw.Do("// Introduced is an autogenerated function, returning the Kubernetes release in which the API struct was introduced and int versions of major and minor for comparison.\n", args)
+		sw.Do("// It is controlled by \""+introducedTagName+"\" tags in types.go.\n", args)
+		sw.Do("func (in *$.type|intrapackage$) Introduced() (string, int64, int64) {\n", args)
+		sw.Do("    return \"$.introduced$\", $.introducedMajor$, $.introducedMinor$\n", args)
+		sw.Do("}\n\n", nil)
+	}
+	if versionedMethodOrDie("Deprecated", t) == nil {
+		sw.Do("// Deprecated is an autogenerated function, returning the Kubernetes release in which the API struct was or will be deprecated and int versions of major and minor for comparison.\n", args)
+		sw.Do("// It is controlled by \""+deprecatedTagName+"\" tags in types.go or  \""+introducedTagName+"\" plus three minor.\n", args)
+		sw.Do("func (in *$.type|intrapackage$) Deprecated() (string, int64, int64) {\n", args)
+		sw.Do("    return \"$.deprecated$\", $.deprecatedMajor$, $.deprecatedMinor$\n", args)
+		sw.Do("}\n\n", nil)
+	}
+	if versionedMethodOrDie("Removed", t) == nil {
+		sw.Do("// Removed is an autogenerated function, returning the Kubernetes release in which the API struct was or will be removed and int versions of major and minor for comparison.\n", args)
```

Comment:
```suggestion
		sw.Do("// Removed is an autogenerated function, returning the release in which the API struct is no longer served, and int versions of major and minor for comparison.\n", args)
```

Reply:
distinguish "no longer served" from "API struct type is removed"? also, if this will generate functions for arbitary API servers' types, the major/minor version might not be a Kubernetes release version, right?

Reply:
is there a reason not to just return the major/minor int versions and drop the string retval?

Reply:
(comments apply to all methods)